juju/terraform-provider-juju
# fix(application) wait for application to be destroyed

## Description

Wait for an application to actually be destroyed when calling Destroy(). This is an issue where terraform is calling Destroy() then Create() when the former completes.  Juju may not have had time to fully destroy the application before we try to recreate it. 

A bit of refactoring to put errors in the same place and use errors.Join() rather than a home grown version.

Stop using 2 versions of the juju/names package, only one should be.

Fixes: #521 

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Other (please describe)

## QA steps

Manual QA steps should be done to test this PR.

```tf
provider juju {}
resource "juju_model" "this" {
  name = "unit-test"
}

resource "juju_application" "githubrunner-29" {
  name  = "github-runner"
  model = juju_model.this.name

  charm {
    name     = "github-runner"
    channel  = "latest/edge"
    base = "ubuntu@22.04"
  }
  units = 1

  storage_directives = {
    runner = "2G"
  }
}
```

Use juju status to see the application running. Then edit the plan to change storage_directive from `2G` to `1G` and re-run the plan.

## Additional notes

JUJU-6348
